### PR TITLE
feat: get & list channel

### PR DESCRIPTION
**Pull request by Lanture1064**
#2

### CODE REVIEW

**Comment by bjwswang** [comment]
@Lanture1064 Any progress?

File: pkg/channel/get.go [hunk 1]
```diff
@@ -17,18 +17,122 @@ limitations under the License.
...
-	//defaultPrintFlag := get.NewGetPrintFlags()
+	
+	defaultPrintFlag := get.NewGetPrintFlags()
+
 	cmd := &cobra.Command{
-		Use: "",
+		Use: "channel [NAME] -n NETWORK-NAME",
 		RunE: func(cmd *cobra.Command, args []string) error {
+			// get config from getter
+			cfg, err := clientcmd.BuildConfigFromKubeconfigGetter("", common.InKubeGetter)
```
Comment (0xff-dev) [inline pkg/channel/get.go:45]:
```go
client, err := common.GetDynamicClient()
```

File: pkg/channel/get.go [hunk 2]
```diff
@@ -17,18 +17,122 @@ limitations under the License.
...
+				TypeMeta: v1.TypeMeta{
+					Kind:       "List",
+					APIVersion: "v1",
+				},
+				ListMeta: v1.ListMeta{},
+			}
+			var obj runtime.Object
+			netName, _ := cmd.Flags().GetString("network")
+
+			if netName == "" {
```
Comment (0xff-dev) [inline pkg/channel/get.go:65]: Set network as a required parameter.
```go
_ = cmd.MarkFlagRequired("network")
```

File: pkg/channel/get.go [hunk 3]
```diff
@@ -17,18 +17,122 @@ limitations under the License.
...
+					list.Items = append(list.Items, runtime.RawExtension{Object: ch})
+				}
+
+			} else {
+				// get channel by name
+				for _, arg := range args {
+					org, err := cli.Resource(schema.GroupVersionResource{
+						Group:    common.IBPGroup,
+						Version:  common.IBPVersion,
+						Resource: common.OrganizationResource,
```
Comment (0xff-dev) [inline pkg/channel/get.go:103]: Channel ?

File: pkg/channel/get.go (hunk 2 above)
Comment (0xff-dev) [reply pkg/channel/get.go:59]: `network` is set as a required parameter, the if judgment is not needed.
How to use the function `MarkFlagRequired` reference https://github.com/bestchains/bc-cli/blob/main/pkg/endorsepolicy/get.go#L174
setting `MarkFlagRequired` in RunE does not take effect.

File: pkg/channel/get.go [hunk 4]
```diff
@@ -17,18 +17,116 @@ limitations under the License.
...
+					list.Items = append(list.Items, runtime.RawExtension{Object: ch})
+				}
+
+			} else {
+				// get channel by name
+				for _, arg := range args {
+					ch, err := cli.Resource(schema.GroupVersionResource{
+						Group:    common.IBPGroup,
+						Version:  common.IBPVersion,
+						Resource: common.NetworkResource,
```
Comment (0xff-dev) [inline pkg/channel/get.go:97]: Resource: common.Channel ?

**Comment by 0xff-dev** [comment]
rebase commits.

File: pkg/channel/get.go [hunk 5]
```diff
@@ -17,18 +17,117 @@ limitations under the License.
...
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/kubectl/pkg/cmd/get"
+
+	"github.com/bestchains/bc-cli/pkg/common"
 )
 
+// NewChanGetCmd returns the `kubectl get` command for channel.
 func NewChanGetCmd(option common.Options) *cobra.Command {
```
Comment (0xff-dev) [inline pkg/channel/get.go:35]: Add this function to the `cmd/bc-cli/get/get.go`.

**Comment by 0xff-dev** [comment]
Submit local run results

File: pkg/channel/get.go [hunk 6]
```diff
@@ -17,18 +17,117 @@ limitations under the License.
...
+			if err != nil {
+				fmt.Fprintln(option.ErrOut, err)
+				return err
+			}
+			_ = p.PrintObj(obj, option.Out)
 			return nil
 		},
 	}
 
+	_ = cmd.MarkFlagRequired("network")
```
Comment (0xff-dev) [inline pkg/channel/get.go:129]:
```go
cmd.Flags().StringP("network", "n", "", "network of the desired channel")
_ = cmd.MarkFlagRequired("network")
```
After adding the flags, then set it as a  required parameter.  
When set as a required parameter, network must not be empty.  we can delete the if statement in line 59.

Comment (bjwswang) [reply pkg/channel/get.go:129]: @Lanture1064 have you tested this CLI commands before you make this PR ?

**Comment by Lanture1064** [comment]
Local test result: 
getChan-localRes

**Review by 0xff-dev — APPROVED** [review]

**Review by Abirdcfly — APPROVED** [review]

**Pull request merged**